### DISCUSSION TITLE
Support disabling tests

### DISCUSSION THREAD

**Discussion by kawazoe** [vitest-dev/vitest#1992] [Ideas]
Right now, vitest doesn't support `xit` form to disable a test or `xdescribe` to disable a whole group. My testing methodology often leads me to write a bunch of empty tests as reminders of what I should be implementing next. With other testing framework, I usually flag them as disabled until I actually implement the tests. This highlight them in test reports which I find quite useful to get a sense of my progress. It's a little something I miss in vitest and the vitest ui.

**Comment by tony19** [comment]
`xit` and `xdescribe` are just aliases for [`test.skip`](https://jestjs.io/docs/api#testskipname-fn) and [`describe.skip`](https://jestjs.io/docs/api#describeskipname-fn) respectively, which Vitest supports already:

 * [`test.skip`](https://vitest.dev/api/#test-skip)
 * [`describe.skip`](https://vitest.dev/api/#describe-skip)

**Reply by kawazoe** [reply]
Thanks! I don't know how I missed that.

**Reply by pvinis** [reply]
Will vitest support the actual `xit` and `fit` aliases? they are more convenient than `test.skip` etc.

**Reply by sheremet-va** [reply]
No. If you want to alias it, you can do so yourself in the setup file:

```ts
import { it, describe } from 'vitest'
globalThis.xit = it.skip
globalThis.xdescribe = describe.skip
```

**Reply by pvinis** [reply]
nice, cool! this should be on the docs somewhere.

**Comment by bburns** [comment]
Supporting xdescribe and xit makes sense, as that's Jest syntax - 

> describe.skip(name, fn)
> Also under the alias: xdescribe(name, fn)
https://jestjs.io/docs/api#describeskipname-fn

But in the meantime, this works - 

vitest.config.js:

```js
export default {
  test: {
    setupFiles: ['./vitest.setup.js'],
  },
}
```

vitest.setup.js:

```js
import { it, describe } from 'vitest'

globalThis.xit = it.skip
globalThis.xdescribe = describe.skip
```

Note: you can't import from vitest in vitest.config.js, as you get this error - 

> Error: Vitest failed to access its internal state.
> One of the following is possible:
> - "vitest" is imported directly without running "vitest" command
> - "vitest" is imported inside "globalSetup" (to fix this, use "setupFiles" instead, because "globalSetup" runs in a different context)

**Reply by Hebilicious** [reply]
You can also do this at the top of each test file

```ts
const xit = it.skip
const xdescribe = describe.skip
```
